core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectionDiagnosticProcessor.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.deployment.steps;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+
+/**
+ * writes a list of all reflective classes to META-INF, if the quarkus.reflection.debug system property is set
+ * <p>
+ * This is not really user facing config for now, which is why it is just a system property instead of using the standard
+ * config mechanism
+ */
+public class ReflectionDiagnosticProcessor {
+
+    @BuildStep
+    public GeneratedResourceBuildItem writeReflectionData(List<ReflectiveClassBuildItem> classes) {
+        if (Boolean.getBoolean("quarkus.reflection.debug")) {
```

Comment:
Should we try to be consistent with the debug system properties naming convention?

The previous one has a totally different naming scheme: https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/extension-authors-guide.adoc#91-saving-application-generated-classes-to-disk .

I don't care which one we choose but we should try to be consistent.

I think `quarkus.debug.<something>` would make sense (and we would reserve the entire `quarkus.debug.` namespace for us).